src/libraries/System.Text.Json/src/System/Text/Json/Nodes/JsonValue.cs

```diff
@@ -14,6 +14,7 @@ namespace System.Text.Json.Nodes
     public abstract partial class JsonValue : JsonNode
     {
         internal const string CreateUnreferencedCodeMessage = "Creating JsonValue instances with non-primitive types is not compatible with trimming. It can result in non-primitive types being serialized, which may have their members trimmed.";
+        internal const string CreateDynamicCodeMessage = "Creating JsonValue instances with non-primitive types is not compatible with runtime code generation. Use System.Text.Json source generation for native AOT applications.";
```

Comment:
"is not compatible with runtime code generation" seems odd to me. What about "requires dynamically generating code at runtime"?

Reply:
Also, the suggestion of "Use System.Text.Json source generation for native AOT applications." sort of doesn't apply here. This API is for JSON DOM manipulation - i.e. building up the object hierarchy in memory. You can't really solve that with source generation.